jsapp/js/account/organizations/requireOrgOwner.component.tsx

```diff
@@ -1,30 +1,34 @@
-import React, {Suspense, useContext, useEffect} from 'react';
+import React, {Suspense, useEffect} from 'react';
 import {useNavigate} from 'react-router-dom';
-import {OrganizationContext} from 'js/account/organizations/useOrganization.hook';
+import {useOrganizationQuery} from 'js/query/queries/organizations.query';
 import LoadingSpinner from 'js/components/common/loadingSpinner';
 import {ACCOUNT_ROUTES} from 'js/account/routes.constants';
+import {ReactQueryStatuses} from 'jsapp/js/constants';
 
 interface Props {
   children: React.ReactNode;
   redirect?: boolean;
 }
 
 export const RequireOrgOwner = ({children, redirect = true}: Props) => {
-  const [organization, _, orgStatus] = useContext(OrganizationContext);
+  const {
+    data: organizationData,
+    status: organizationStatus,
+  } = useOrganizationQuery();
   const navigate = useNavigate();
 
   useEffect(() => {
     if (
       redirect &&
-      !orgStatus.pending &&
-      organization &&
-      !organization.is_owner
+      !(organizationStatus === ReactQueryStatuses.pending) &&
```

Comment:
I think we should just use `'pending'` rather than importing QueryStatus here

Reply:
@p2edwards That's OK with me. Using QueryStatus does give us a bit more type safety in the event they ever decide to change their status names, but presumably we'd see that in a breaking change note when updating the package.